### ISSUE TITLE
[package] poco/1.9.4 to 1.11.0: cannot use NetSSL component with OpenSSL on Windows

### ISSUE THREAD

**Issue by jngrb** [conan-io/conan-center-index#6311]
Use-case: I want to be able to consume Poco to build an SSL-enabled web server using the OpenSSL library on any platform.

Bug: since a few changes were made to the Poco recipe this year, this use-case is no longer supported. As of now, only WinTLS is supported by the CCI poco recipe on Windows (while plain Poco supports either OpenSSL or WinTLS on Windows).

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **poco/1.8.1 to 1.11.0**
  * Operating System+version: **Windows**
  * Compiler+version: **Visual Studio**
  * Conan version: **conan 1.35.1**
  * Python version: **Python 3.7.3**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=15
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
Try to consume poco (any version, currently available are Poco v1.8.1 to v1.11.0) with enable_netssl=True and enable_netssl_win=False to have **SSL support using the OpenSSL** on Windows (instead of Windows' TLS).

Before https://github.com/conan-io/conan-center-index/pull/4205 and https://github.com/conan-io/conan-center-index/commit/6ea152e89a6b2303739f59ebfc4d1e7e227bb8a3 were implemented, I could do this using:

- enable_netssl=True
- enable_netssl_win=False
- force_openssL=True

on Windows

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
ERROR: poco/1.9.4: option 'enable_netssl' doesn't exist
Possible options are ['shared', 'enable_apacheconnector', 'enable_cppparser', 'enable_crypto', 'enable_data', 'enable_data_odbc', 'enable_data_sqlite', 'enable_encodings', 'enable_json', 'enable_mongodb', 'enable_net', 'enable_netssl_win', 'enable_pdf', 'enable_pagecompiler', 'enable_pagecompiler_file2page', 'enable_pocodoc', 'enable_redis', 'enable_sevenzip', 'enable_util', 'enable_xml', 'enable_zip']
```

</details>

**Comment by jngrb** [comment]
@mathbunnyru , what was the intention behind the removal of the NetSSL option for Windows in #4205 ?

**Comment by mathbunnyru** [comment]
> @mathbunnyru , what was the intention behind the removal of the NetSSL option for Windows in #4205 ?

This is the issue I was fixing (and I also had).

https://github.com/conan-io/conan-center-index/issues/4184

**Comment by jngrb** [comment]
> This is the issue I was fixing (and I also had).
> 
> #4184

@mathbunnyru , I think it is a valid use case to use NetSSL on Windows (instead of NetSSL_win). You removed that option.
Did you encounter any issue with NetSSL on Windows? Because I didn't. In my projects (with a non-CCI Poco library), both options work fine (OpenSSL or WinTLS).

In my opinion, the only issue is the case where both NetSSL options are set simultaneous. That should be prevented.

See the section "Poco special build parameters" in [the Poco Getting Started Giude](https://pocoproject.org/docs/00200-GettingStarted.html).

> Here an overview of POCO build options:
> [...]
> ENABLE_NETSSL Set to OFF|ON (default is ON) to build NetSSL support library (Need installed openssl libraries)
> [...]
> Windows only parameter:
> [...]
> ENABLE_NETSSL_WIN Set to OFF|ON (default is OFF) to build NetSSL support library(Need installed openssl libraries For Windows only)

=> enableNetSSL is a valid Option for Windows (and only mutually exclusive with enableNetSSL_win).

So why did you remove it?

**Comment by jngrb** [comment]
@mathbunnyru , I just read your [comment ](https://github.com/conan-io/conan-center-index/issues/4184#issuecomment-757505218) on #4184

> Somehow there are two libraries PocoNetSSLWinmdd.lib and PocoNetSSLmdd.lib which exist simultaneously on Windows, share a lot of symbols and that causes link errors.
> I think the error is how Poco is built in Conan under Windows.

As you point out the problem is can both NetSSLWin and NetSSL  are build leading to linker errors.
But the user should have the option to use either of the two.

So instead of
```
    def config_options(self):
        if self.settings.os == "Windows":
            [...]
            del self.options.enable_netssl
```
we should just prevent the user from setting both to true:
```
    def validate(self):
	[...]
        if self.options.enable_netssl and self.options.get_safe("enable_netssl_win", False):
            raise ConanInvalidConfiguration("Conflicting enable_netssl[_win] settings")
```

That's what I suggest in my PR #6314.

**Comment by jngrb** [comment]
closed because the PR #6314 with the changed default values for the NetSSL and NetSSL_win options was merged to master
